lib/Sema/TypeCheckAttr.cpp

```diff
@@ -654,6 +654,7 @@ bool AttributeEarlyChecker::visitAbstractAccessibilityAttr(
 
   // Or within protocols.
   if (isa<ProtocolDecl>(D->getDeclContext())) {
+    diagnoseAndRemoveAttr(attr, diag::access_control_in_protocol_detailed_info);
```

Comment:
You're going to want to emit the note thru the type checker instead of this little convenience function, or you might pop this fixit twice.  Also, emit notes after primary diagnostics.